apps/emqx/test/emqx_packet_tests.erl

```diff
@@ -85,7 +80,7 @@ invalid_utf8_fallback_test() ->
     ]),
     Lost = size(Payload) - iolist_size(Prefix),
     Expected = iolist_to_binary([
-        "hex:", binary:encode_hex(Prefix), "...(", integer_to_list(Lost), " bytes)"
+        binary:encode_hex(Prefix), "...(", integer_to_list(Lost), " bytes)"
```

Comment:
should keep the `"hex:"` prefix

Reply:
I reverted your change https://github.com/emqx/emqx/pull/13909 in back to `Payload(hex)=7B22417274697374223A202261222C22536F6E675469746C65223A202261227D`. See the log msg examples above.

Reply:
Now the `emqx_packet:format_payload/2` returns the final encoding it uses, and I also added an additional metadata `payload_encode` for that.